code/src/test/java/habit_mode/test/model/habit_manager/TestAdd.java

```diff
@@ -0,0 +1,95 @@
+package habit_mode.test.model.habit_manager;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.Test;
+
+import habit_mode.model.Frequency;
+import habit_mode.model.Habit;
+import habit_mode.model.HabitManager;
+
+class TestAdd {
+    @Test
+	void testAddWithValidHabit() {
+		Habit habit = new Habit("Hello!", Frequency.DAILY);
+        HabitManager manager = new HabitManager();
+
+        boolean result = manager.add(habit);
+
+        assertAll(
+            () -> {assertTrue(result, "Check if return is correct");},
+            () -> {assertTrue(manager.contains(habit), "Check if habit was added");}
+        );
+	}
```

Comment:
Uses both tabs and spaces. Go through your files and make sure that you aren't using any tab characters.